embedded-tests/src/test/java/org/apache/druid/testing/embedded/indexing/autoscaler/CostBasedAutoScalerIntegrationTest.java

```diff
@@ -161,6 +155,20 @@ public void test_autoScaler_computesOptimalTaskCountAndProduceScaleDown()
     cluster.callApi().postSupervisor(spec.createSuspendedSpec());
   }
 
+  /**
+   * Tests that scale down happen during task rollover via checkTaskDuration().
+   *
+   * <p>Test flow:</p>
+   * <ol>
+   *   <li>Start supervisor with 10 tasks and 50 partitions, minimal data (500 records)</li>
+   *   <li>Wait for initial tasks to start running</li>
+   *   <li>Wait for the first task rollover to complete (task duration is 8 seconds)</li>
+   *   <li>Verify that after rollover, fewer tasks are running due to cost-based autoscaler (no ingestion at all)</li>
+   * </ol>
+   *
+   * <p>Scale down during rollover is triggered in {@code SeekableStreamSupervisor.checkTaskDuration()}
+   * when all task groups have rolled over and the autoscaler recommends a lower task count.</p>
+   */
```

Comment:
I feel we should omit the javadoc. The test itself should be readable enough to follow through the details.